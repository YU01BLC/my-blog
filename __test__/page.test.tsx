import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Page from '../src/app/page';

describe('PageコンポーネントSnapshot', () => {
  it('snapshot', () => {
    // Pageコンポーネントのレンダリング結果をオブジェクトとして定義(asFragment: 一般的にスナップショットの取得結果であることを示す名称として用いられる)
    const { asFragment } = render(<Page />);
    // asFragment()で取得したスナップショットを現在の状態と比較する
    expect(asFragment()).toMatchSnapshot();
  });
});
describe('Pageコンポーネント表示確認', () => {
  it('テキスト「記事タイトル」が表示されること', () => {
    // Pageコンポーネントをレンダリング
    render(<Page />);
    // 「記事タイトル(dummy)」というテキストを含む要素を取得
    const headElement = screen.getByText('記事タイトル(dummy)');
    // 取得した上記要素が画面上に存在することを確認
    expect(headElement).toBeInTheDocument();
  });
});
