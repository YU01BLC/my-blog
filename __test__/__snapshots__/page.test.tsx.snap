// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageコンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-he2f3o"
    data-testid="article"
  >
    <div
      class="MuiBox-root css-1b420e2"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        data-testid="dummy-article"
      >
        <div
          class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              記事タイトル(dummy)
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
            >
              2023/12/1
            </span>
          </div>
        </div>
        <img
          alt="blog image"
          class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
          height="500"
          src="https://source.unsplash.com/collection/1346951/1000x500?sig=2"
          width="100%"
        />
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
          >
            ダミー記事です。
            <br />
            右上の「記事を書く」ボタンから新規記事を執筆してください。
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
            data-testid="dummy-article--text"
          >
            ※   記事投稿後ダミー記事は非表示となります。
          </p>
          <div
            class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
          >
            <div
              aria-expanded="false"
              aria-label="show more"
              class="MuiBox-root css-d3fn1e"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                続きを読む
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-0"
        data-testid="article-list"
      >
        <a
          href="/hoge"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  記事タイトル: 1
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
                >
                  2023/12/1
                </span>
              </div>
            </div>
            <img
              alt="blog image: 1"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="500"
              src="https://source.unsplash.com/collection/1346951/1000x500?sig=1"
            />
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
                data-testid="article-list--text1"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  記事タイトル: 2
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
                >
                  2023/12/1
                </span>
              </div>
            </div>
            <img
              alt="blog image: 2"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="500"
              src="https://source.unsplash.com/collection/1346951/1000x500?sig=2"
            />
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
                data-testid="article-list--text2"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
              </p>
            </div>
          </div>
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-5y2sye-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
                >
                  記事タイトル: 3
                </span>
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
                >
                  2023/12/1
                </span>
              </div>
            </div>
            <img
              alt="blog image: 3"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="500"
              src="https://source.unsplash.com/collection/1346951/1000x500?sig=3"
            />
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-136xlp0-MuiTypography-root"
                data-testid="article-list--text3"
              >
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Veritatis quod maiores totam aperiam necessitatibus est, exercitationem architecto quibusdam accusamus maxime asperiores eaque adipisci itaque nemo, minus odio, alias dolor possimus!
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
